### PR TITLE
Address todos in stream dev guides

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Address todos in standalone stream on RabbitMQ
 - Address todos in standalone stream on Kafka
 - Address todos in data flow streams
 - Address todos in continuous delivery of apps

### CODE REVIEW

File: content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md [hunk 1]
```diff
@@ -91,29 +100,89 @@ If you haven't downloaded the completed source example, you will need to perform
 
     ```
 
-**TODO some discussion of the annotations should be made**
+In the above application, the `@EnableBinding` annotation lets you bind your `Source` application's `output` to the configured `RabbitMQ` exchange using the RabbitMQ binder.
+The `@EnableScheduling` annotation invokes `sendEvents` with the configured fixed delay `1` second.
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md]: > The `@EnableScheduling` annotation invokes `sendEvents` function with the configured fixed delay of `1` second.

File: content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md [hunk 2]
```diff
@@ -233,79 +350,267 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
...
+We can individually run these custom applications before creating a pipeline using Spring Cloud Data Flow.
+To run, we need to explicitly set the Spring Cloud Stream bindings destination property and the server port for accessing the actuator endpoints:
 
 ```
 spring.cloud.stream.bindings.output.destination=test-usage-detail
+server.port=9001
 ```
 
 In this case, we can use some test RabbitMQ `exchanges` to verify the outbound and inbound messages.
 For instance, you can set the `output` binding to a test RabbitMQ exchange `test-usage-detail` and see if the messages get posted to the exchange.
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md]: > For instance, you can set the `output` binding to a test RabbitMQ exchange `test-usage-detail` and verify whether the messages get published to the exchange.

File: content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md [hunk 3]
```diff
@@ -233,79 +350,267 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
...
+Also, if you want the `UsageCostProcessor` application's output message delivery guarantee, then you need to set the `requiredGroups` for the `output` binding:
+
+```
+spring.cloud.stream.bindings.output.producer.requiredGroups=usage-cost
+```
+
+Now, you can run the `UsageCostProcessor` application as:
+
+```
+java -jar target/usage-cost-processor-rabbit-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-detail --spring.cloud.stream.bindings.output.destination=test-usage-cost --server.port=9002 --spring.cloud.stream.bindings.input.group=usage-detail --spring.cloud.stream.bindings.output.producer.requiredGroups=usage-cost &
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md]: Perhaps it is better to remove `&` from the command. Let the user see the console output as opposed to running it in the background.

File: content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md [hunk 4]
```diff
@@ -233,79 +350,267 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
...
 ```
 spring.cloud.stream.bindings.input.destination=test-usage-cost
+server.port=9003
 ```
 
 You can run the standalone `UsageCostLogger` sink application as,
 
 ```
-java -jar target/usage-cost-logger-rabbit-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-cost &
+java -jar target/usage-cost-logger-rabbit-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-cost --server.port=9003 &
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/1-standalone-stream-rabbitmq.md]: Remove `&` from every command

File: content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md [hunk 5]
```diff
@@ -278,33 +353,175 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
...
 
-#### Running the Sink
+The source application `UsageDetailSender`'s output is connected to the `UsageCostProcessor` processor application's input.
+The `UsageCostProcessor` application's output is connected to the `UsageCostLogger` sink application's input.
 
-You can run the standalone `UsageCostLogger` sink application as,
+Since we use Kafka as the streaming platform, we have selected Spring Cloud Stream `Kafka` binder for this application.
+When these applications are run, the `Kafka` binder binds the applications output/input boundaries into the corresponding destinations (topics) at Kafka.
+
+Let's choose our destinations at Kafka:
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md]: > Let's choose our destinations for Kafka:

File: content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md [hunk 6]
```diff
@@ -278,33 +353,175 @@ Now let's build the Usage Cost Logger application. In the directory `usage-cost-
...
+After unpacking the downloaded archive, you can start `ZooKeeper` and `Kafka` servers as follows:
 
 ```
-java -jar target/usage-cost-logger-kafka-0.0.1-SNAPSHOT.jar --spring.cloud.stream.bindings.input.destination=test-usage-cost &
+./bin/zookeeper-server-start.sh config/zookeeper.properties &
 ```
 
-Now, you can see that this application logs the usage cost detail.
+```
+./bin/kafka-server-start.sh config/server.properties &
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md]: I'm still not a fan of background processes. It shouldn't be hidden from the users who are trying it - better to see all the console output.

File: content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md [hunk 7]
```diff
@@ -539,4 +756,4 @@ kubectl delete all -l app=kafka
 
 ## Testing
 
-**TODO We did not cover testing, in order to get to the heart of the matter quickly. We can circle back and show how to create unit test for spring cloud stream apps.**
+**TODO** **We did not cover testing, in order to get to the heart of the matter quickly. We can circle back and show how to create unit test for spring cloud stream apps.**
```
Comment (sabbyanandan) [inline content/documentation/pages/3-stream-developer-guides/2-streams/2-standalone-stream-kafka.md]: Should we remove this TODO?

**Comment by markpollack** [comment]
edited, added some TODOs and merged as 1e9d9505144a2da7e13bb5f6d8a8f998922b035e

**Pull request closed without merging**